chalk-engine/src/context.rs

```diff
@@ -115,10 +115,10 @@ pub trait Context: Clone + Debug {
     ) -> Self::GoalInEnvironment;
 
     /// Upcast this domain goal into a more general goal.
-    fn into_goal(domain_goal: Self::DomainGoal) -> Self::Goal;
+    fn into_goal(&self, domain_goal: Self::DomainGoal) -> Self::Goal;
```

Comment:
Why don't we just move these methods onto the `InferenceTable` trait? They only seem to be used from `simplify.rs`, and it has an `InferenceTable`. That gives us access to everything we need, no?

Reply:
Yes this is simpler indeed, I blindly wanted to somehow preserve the semantics of the `Context` trait :)